### DISCUSSION TITLE
`cursor`'s cask code seems inconsistent with official downloads

### DISCUSSION THREAD

**Discussion by erazemk** [Homebrew/discussions#5800] [Tap maintenance and Homebrew development]
### Output of `brew config`

```shell
HOMEBREW_VERSION: 4.4.11
ORIGIN: https://github.com/Homebrew/brew
HEAD: ad356d365837a691a52dfec32b3fb5f282f3aefc
Last commit: 2 days ago
Branch: stable
Core tap JSON: 11 Dec 11:52 UTC
Core cask tap JSON: 11 Dec 11:52 UTC
HOMEBREW_PREFIX: /opt/homebrew
HOMEBREW_CASK_OPTS: []
HOMEBREW_EDITOR: hx
HOMEBREW_MAKE_JOBS: 12
HOMEBREW_NO_ENV_HINTS: set
Homebrew Ruby: 3.3.6 => /opt/homebrew/Library/Homebrew/vendor/portable-ruby/3.3.6/bin/ruby
CPU: dodeca-core 64-bit arm_blizzard_avalanche
Clang: 16.0.0 build 1600
Git: 2.47.1 => /opt/homebrew/bin/git
Curl: 8.7.1 => /usr/bin/curl
macOS: 15.1.1-arm64
CLT: 16.1.0.0.1.1729049160
Xcode: 16.1
Rosetta 2: false
```

### Output of `brew doctor`

```shell
Your system is ready to brew.
```

### Description of issue

I randomly decided to read [Cursor](https://www.cursor.com)'s cask code and I saw that the URLs are not the same as the ones from the official downloads. I see that the official download is an installer and I haven't yet checked out what URLs it uses to download the app, but is it possible that the links are outdated but just happen to still redirect properly?

For sure at least the [homepage link](https://github.com/Homebrew/homebrew-cask/blob/master/Casks/c/cursor.rb#L12) is incorrect, but if you let me know based on what the other URLs were set, I can prepare a PR with the relevant fixes.

**Comment by p-linnane** [comment]
The homepage in the cask redirects to cursor.com. This is not the first time they have changed their homepage.

```
╰─ curl -IL cursor.sh
HTTP/1.0 308 Permanent Redirect
Content-Type: text/plain
Location: https://cursor.sh/
Refresh: 0;url=https://cursor.sh/
server: Vercel

HTTP/2 308
cache-control: public, max-age=0, must-revalidate
content-type: text/plain
date: Wed, 11 Dec 2024 20:03:59 GMT
location: https://www.cursor.com/
refresh: 0;url=https://www.cursor.com/
server: Vercel
strict-transport-security: max-age=63072000
x-vercel-id: sfo1::4lr5m-1733947439807-3ed0b9c85f9b
```

The URL for the binaries (beginning with `https://download.todesktop.com/230313mzl4w4u92`) is also what is used [since the cask was added.](https://github.com/Homebrew/homebrew-cask/commit/44b5b875300130d43fce7bee9518cdb7ed83c163)

**Reply by erazemk** [reply]
Incorrect was perhaps a bit harsh, I meant more it's not their main page.
